pkg/customunit/client.go

```diff
@@ -0,0 +1,145 @@
+package customunit
+
+import (
+	"context"
+	"time"
+
+	"github.com/Axway/agent-sdk/pkg/agent/cache"
+	cu "github.com/Axway/agent-sdk/pkg/amplify/agent/customunits"
+	"github.com/Axway/agent-sdk/pkg/util/log"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+type customUnitClient struct {
+	logger    log.FieldLogger
+	quotaInfo *cu.QuotaInfo
+	dialOpts  []grpc.DialOption
+	cOpts     []grpc.CallOption
+	url       string
+	conn      *grpc.ClientConn
+	isRunning bool
+	cache     cache.Manager
```

Comment:
same as my other comment, lets use a local interface with just the funcs we need from the cache manager